include/volk/volk_avx2_intrinsics.h

```diff
@@ -345,4 +345,46 @@ static inline void vector_32fc_index_min_variant1(__m256 in0,
     *current_indices = _mm256_add_epi32(*current_indices, indices_increment);
 }
 
+/*
+ * Approximate sin(x) via polynomial expansion
+ * on the interval [-pi/4, pi/4]
+ *
+ * Maximum absolute error ~7.3e-9
+ * sin(x) = x + x^3 * (s1 + x^2 * (s2 + x^2 * s3))
+ */
+static inline __m256 _mm256_sin_poly_avx2(const __m256 x)
+{
+    const __m256 s1 = _mm256_set1_ps(-0x1.555552p-3f);
```

Comment:
Can you add a reference to the approximation? E.g. a paper where the approach is discussed. Out how you obtained these exact values?

Reply:
I use the Sollya Remez algorithm for minimax polynomial calculations and the approach outlined here: https://basesandframes.wordpress.com/wp-content/uploads/2020/04/even_faster_math_functions_green_2020.pdf

It's not strictly known what is the best, in terms of speed and accuracy, approximation of many special functions when considering the fact that functions like reciprocals, square roots and fma instructions exist and their cost in terms of latency and cycles. It remains an active area of research, see for example https://pavpanchekha.com/blog/synthesize-range-reductions.html

Reply:
If you could add the links and your quick explanation in a comment next to your code, that'd be great. 
If I imagine myself trying to figure out the details again in a year or so, that'd be extremely helpful.